File: app/src/main/java/in/woloo/www/period_tracker/model/GetPeriodDataResponse.java
package in.woloo.www.period_tracker.model;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class GetPeriodDataResponse{

	@SerializedName("code")
	private String code;

	@SerializedName("data")
	private Data data;

	@SerializedName("message")
	private String message;

	@SerializedName("status")
	private String status;

	public String getCode(){
		return code;
	}

	public Data getData(){
		return data;
	}

	public String getMessage(){
		return message;
	}

	public String getStatus(){
		return status;
	}

	public class Data{

		@SerializedName("period_date")
		private String periodDate;

		@SerializedName("cycle_lenght")
		private int cycleLenght;

		@SerializedName("updated_at")
		private String updatedAt;

		@SerializedName("user_id")
		private int userId;

		@SerializedName("log")
		private Log log;

		@SerializedName("created_at")
		private String createdAt;

		@SerializedName("id")
		private int id;

		@SerializedName("period_length")
		private int periodLength;

		@SerializedName("luteal_length")
		private String lutealLength;

		public String getPeriodDate(){
			return periodDate;
		}

		public int getCycleLenght(){
			return cycleLenght;
		}

		public String getUpdatedAt(){
			return updatedAt;
		}

		public int getUserId(){
			return userId;
		}

		public Log getLog(){
			return log;
		}

		public String getCreatedAt(){
			return createdAt;
		}

		public int getId(){
			return id;
		}

		public int getPeriodLength(){
			return periodLength;
		}

		public String getLutealLength(){
			return lutealLength;
		}
	}
}

File: app/src/main/java/in/woloo/www/more/period_tracker/model/GetPeriodDataResponse.kt
package `in`.woloo.www.more.period_tracker.model

import com.google.gson.annotations.SerializedName

class GetPeriodDataResponse {
    @SerializedName("code")
    val code: String? = null

    @SerializedName("data")
    val data: Data? = null

    @SerializedName("message")
    val message: String? = null

    @SerializedName("status")
    val status: String? = null

    inner class Data {
        @SerializedName("period_date")
        val periodDate: String? = null

        @SerializedName("cycle_lenght")
        val cycleLenght: Int = 0

        @SerializedName("updated_at")
        val updatedAt: String? = null

        @SerializedName("user_id")
        val userId: Int = 0

        @SerializedName("log")
        val log: Log? = null

        @SerializedName("created_at")
        val createdAt: String? = null

        @SerializedName("id")
        val id: Int = 0

        @SerializedName("period_length")
        val periodLength: Int = 0

        @SerializedName("luteal_length")
        val lutealLength: String? = null
    }
}